test/NotificationActivationTests/APITests.cpp

```diff
@@ -0,0 +1,129 @@
+        {
+            TP::RemovePackage(GetTestPackageFullName());
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            return true;
+        }
+
+        TEST_METHOD_CLEANUP(MethodUninit)
+        {
+            return true;
+        }
+
+        TEST_METHOD(AppNotificationForeground)
+        {
+            BEGIN_TEST_METHOD_PROPERTIES()
+                TEST_METHOD_PROPERTY(L"RunAs", L"UAP")
+                TEST_METHOD_PROPERTY(L"UAP:AppxManifest", L"NotificationActivation-AppxManifest.xml")
+                END_TEST_METHOD_PROPERTIES();
+
+            bool notificationReceived{};
+            winrt::event_token appNotificationToken{ AppNotificationManager::Default().NotificationInvoked([&](const auto&, AppNotificationActivatedEventArgs const& toastArgs)
+            {
+                winrt::hstring argument{ toastArgs.Argument() };
+                // IMap<hstring, hstring> userInput { toastArgs.UserInput() };
+                VERIFY_ARE_EQUAL(argument, c_appNotificationArgument);
+
+                notificationReceived = true;
+            }) };
+
+            AppNotificationManager::Default().Register();
+
+            auto toastActivationCallback{ winrt::create_instance<INotificationActivationCallback>(c_taefAppNotificationComServerId, CLSCTX_ALL) };
+            // Returns after payload delivered to foreground handler
+            VERIFY_SUCCEEDED(toastActivationCallback->Activate(c_appNotificationFakeAUMID.c_str(), c_appNotificationArgument.c_str(), nullptr, 0));
+
+            VERIFY_IS_TRUE(notificationReceived);
+        }
+
+        TEST_METHOD(AppNotificationBackground)
+        {
+            RunTest(L"AppNotificationBackgroundTest", testWaitTime()); // Need to launch one time to enable background activation.
+
+            auto event{ CreateTestEvent(c_testNotificationPhaseEventName) };
+            auto toastActivationCallback{ winrt::create_instance<INotificationActivationCallback>(c_appNotificationComServerId, CLSCTX_ALL) };
+            VERIFY_SUCCEEDED(toastActivationCallback->Activate(c_appNotificationFakeAUMID.c_str(), c_appNotificationArgument.c_str(), nullptr, 0));
+
+            wil::unique_event failed{ CreateTestEvent(c_testFailureEventName) };
+            WaitForEvent(event, failed);
+        }
+
+        TEST_METHOD(PushNotificationForeground)
+        {
+            // Need to incorporate COM server work for PushNotification foreground handling.
+        }
+
+        TEST_METHOD(PushNotificationBackground)
+        {
+            RunTest(L"PushNotificationBackgroundTest", testWaitTime()); // Need to launch one time to enable background activation.
+
+            auto event{ CreateTestEvent(c_testNotificationPhaseEventName) };
+
+            auto localBackgroundTask{ winrt::create_instance<winrt::Windows::ApplicationModel::Background::IBackgroundTask>(c_pushNotificationComServerId, CLSCTX_ALL) };
```

Comment:
nit: using namespace winrt::Windows::ApplicationModel::Background; - :)

Reply:
Yeah, a few names in this file can be shortened.

Reply:
Fixed